src/compiler/checker.ts

```diff
@@ -9558,23 +9862,62 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
             }
 
             function serializeEnum(symbol: Symbol, symbolName: string, modifierFlags: ModifierFlags) {
+                const internalSymbolName = getInternalSymbolName(symbol, symbolName);
+                context.approximateLength += 9 + internalSymbolName.length; // `enum internalName { }`
+                const members: EnumMember[] = [];
+                const memberProps = filter(getPropertiesOfType(getTypeOfSymbol(symbol)), p => !!(p.flags & SymbolFlags.EnumMember));
+                let i = 0;
+                for (const p of memberProps) {
+                    i++;
+                    if (checkTruncationLengthIfExpanding(context) && (i + 2 < memberProps.length - 1)) {
+                        context.out.truncated = true;
+                        members.push(factory.createEnumMember(` ... ${memberProps.length - i} more ... `));
+                        const last = memberProps[memberProps.length - 1];
+                        const initializedValue = last.declarations && last.declarations[0] && isEnumMember(last.declarations[0]) ? getConstantValue(last.declarations[0]) : undefined;
+                        const initializer = initializedValue === undefined ? undefined :
+                            typeof initializedValue === "string" ? factory.createStringLiteral(initializedValue) :
+                            factory.createNumericLiteral(initializedValue);
+                        const memberName = unescapeLeadingUnderscores(last.escapedName);
+                        const member = factory.createEnumMember(
+                            memberName,
+                            initializer,
+                        );
+                        members.push(member);
+                        break;
+                    }
+                    // TODO: Handle computed names
+                    // I hate that to get the initialized value we need to walk back to the declarations here; but there's no
+                    // other way to get the possible const value of an enum member that I'm aware of, as the value is cached
+                    // _on the declaration_, not on the declaration's symbol...
+                    const memberDecl = p.declarations && p.declarations[0] && isEnumMember(p.declarations[0]) ?
+                        p.declarations[0] :
+                        undefined;
+                    let initializer: Expression | undefined;
+                    let initializerLength: number;
+                    if (isExpanding(context) && memberDecl && memberDecl.initializer) {
+                        initializer = visitNode(memberDecl.initializer, factory.cloneNode, isExpression);

```

Comment:
This isn't a deep clone (`cloneNode` isn't a visitor method so doesn't recursively clone), so why isn't this just `factory.cloneNode(memberDecl.initializer)`?

Reply:
If you want a deep clone, there's a `getSynthesizedDeepClone` in `services` that can be moved that does the right thing (quickfixes do this a lot, ofc).